File: backend/schemas/stock_import.py
from typing import Optional

from pydantic import BaseModel, Field


class StockImportLine(BaseModel):
    label: str
    quantite: float
    unite: str
    ingredient_id: Optional[str] = None
    matched: bool = False


class StockImportTextRequest(BaseModel):
    text: str = Field(min_length=1)
    apply: bool = False


class StockImportImageRequest(BaseModel):
    """Import depuis image (base64) et/ou texte OCR déjà extrait."""

    image_base64: Optional[str] = None
    text: Optional[str] = None
    apply: bool = False


class StockImportTextResponse(BaseModel):
    lines: list[StockImportLine]
    applied: int = 0
    source: str = "text"

File: frontend/src/api/stockImport.ts
import { api } from "./http";
import type { IngredientStockOut } from "./stockAlerts";

export type StockImportLine = {
  label: string;
  quantite: number;
  unite: string;
  ingredient_id: string | null;
  matched: boolean;
};

export type StockImportResult = {
  lines: StockImportLine[];
  applied: number;
};

export function importStockText(
  profilId: string,
  token: string,
  payload: { text: string; apply: boolean }
) {
  return api<StockImportResult>(`/stock/${profilId}/import-text`, {
    method: "POST",
    token,
    body: payload,
  });
}

/** Import depuis image base64 (OCR / hook text:…). */
export function importStockImage(
  profilId: string,
  token: string,
  payload: { image_base64?: string; text?: string; apply: boolean }
) {
  return api<StockImportResult & { source?: string }>(
    `/stock/${profilId}/import-image`,
    {
      method: "POST",
      token,
      body: payload,
    }
  );
}

export type { IngredientStockOut };
